### ISSUE TITLE
Mapped object types aren't checked for implicit any errors

### ISSUE THREAD

**Issue by DanielRosenwasser** [Microsoft/TypeScript#21011]
```ts
type Foo = { [P in "hello"] };
```

This is equivalent to

```ts
type Foo = { [P in "hello"]: any };
```

But doesn't trigger a `noImplicitAny` error.

**Comment by mhegazy** [comment]
PRs welcomed.

**Comment by DanielRosenwasser** [comment]
I'd expect this to provide an error like `Mapped object type implicitly has an 'any' template type."

**Comment by jack-williams** [comment]
Would the fix look something like?

```typescript
function checkMappedType(node: MappedTypeNode) {
    checkSourceElement(node.typeParameter);
    checkSourceElement(node.type);
    // -- new code
    if(noImplicitAny && nodeIsMissing(node.type)) {
        error(node, Diagnostics.<SOME_NEW_ERROR_CODE_HERE>);
    }
    // -- end new code
    const type = <MappedType>getTypeFromMappedTypeNode(node);
    const constraintType = getConstraintTypeFromMappedType(type);
    checkTypeAssignableTo(constraintType, stringType, node.typeParameter.constraint);
}
```
Things I'm not sure of:
- Is it better to extend and use `reportImplicitAnyError`?
- Is the error checking code called too early (should it be after fetching the types?)
